external/src/main/java/com/redhat/cloud/policies/engine/actions/plugins/NotificationActionPluginListener.java

```diff
@@ -44,72 +47,107 @@
+        Set<PoliciesAction.Event> events = policiesAction.getEvents();
 
         for (Map.Entry<String, String> tagEntry : actionMessage.getAction().getEvent().getTags().entries()) {
             String value = tagEntry.getValue();
             if (value == null) {
                 // Same behavior as previously with JsonObjectNoNullSerializer with old hooks
                 value = "";
             }
-            payloadBuilder.addTag(tagEntry.getKey(), value);
-        }
 
-        notificationAction.setAccountId(actionMessage.getAction().getTenantId());
+            if (!context.getTags().containsKey(tagEntry.getKey())) {
+                context.getTags().put(tagEntry.getKey(), new HashSet<>());
+            }
+
+            context.getTags().get(tagEntry.getKey()).add(value);
+        }
 
-        // Add the wanted properties here..
         Trigger trigger = actionMessage.getAction().getEvent().getTrigger();
 
-        payloadBuilder.setPolicyId(trigger.getId())
-                .setPolicyName(trigger.getName())
-                .setPolicyDescription(trigger.getDescription());
-
         for (Set<ConditionEval> evalSet : actionMessage.getAction().getEvent().getEvalSets()) {
             for (ConditionEval conditionEval : evalSet) {
                 if (conditionEval instanceof EventConditionEval) {
                     EventConditionEval eventEval = (EventConditionEval) conditionEval;
-                    payloadBuilder.setPolicyCondition(eventEval.getCondition().getExpression())
-                            .setSystemCheckIn(LocalDateTime.from(DateTimeFormatter.ISO_OFFSET_DATE_TIME.parse(eventEval.getContext().get("check_in"))))
-                            .setInsightsId(eventEval.getContext().get("insights_id"))
-                            .setDisplayName(eventEval.getValue().getTags().get("display_name").iterator().next());
 
-                    String name = actionMessage.getAction().getEvent().getTrigger().getName();
-                    String triggerId = actionMessage.getAction().getEvent().getTrigger().getId();
+                    context.setSystemCheckIn(LocalDateTime.from(DateTimeFormatter.ISO_OFFSET_DATE_TIME.parse(eventEval.getContext().get("check_in"))));
+                    context.setInsightsId(eventEval.getContext().get("insights_id"));
+                    context.setDisplayName(eventEval.getValue().getTags().get("display_name").iterator().next());
 
-                    payloadBuilder.addTrigger(triggerId, name);
+                    PoliciesAction.Event event = new PoliciesAction.Event();
+                    event.getPayload().setPolicyCondition(eventEval.getCondition().getExpression());
+                    event.getPayload().setPolicyId(trigger.getId());
+                    event.getPayload().setPolicyName(trigger.getName());
+                    event.getPayload().setPolicyDescription(trigger.getDescription());
+
+                    policiesAction.getEvents().add(event);
+                    break;
```

Comment:
Did you add this `break` because there can only be one `EventConditionEval` instance in the `evalSet`?

Reply:
Yes, there is only one.